### PR TITLE
Dsb/oracle 6.3.0 update

### PR DESCRIPTION
**Pull request by danbeglin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/oracledb/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f98f045f7a923a105701c7335b925131fdeef057
</p>

**Review by sandersn — APPROVED** [review]

**Comment by mfteuscher** [comment]
Possibly look at merging #68930 into this PR before merging to master?

File: types/oracledb/index.d.ts [hunk 1]
```diff
@@ -925,6 +937,21 @@ declare namespace OracleDB {
...
          */
         currentSchema?: string | undefined;
+        /**
+         * This read-only property is a string that specifies the Oracle Database domain name associated with the connection. This property returns the same value as the SQL expression:
+         * SELECT UPPER(VALUE) FROM V$PARAMETER WHERE NAME = 'db_domain';
+         * The above SQL expression returns NULL if the domain name is not specified. The dbDomain property returns an empty string in this case.
+         * 
+         * @since 6.3
+         */
+        readonly dbDomain?: string;
```
Comment (BekYahia) [inline types/oracledb/index.d.ts:947]: It's highly suggested all optional props should've `| undefined` to support `exactOptionalPropertyTypes`. 

https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes

Comment (dannyb648) [reply types/oracledb/index.d.ts:947]: This is really interesting - weve had quite a few people raise PRs and requests direct about these | undefined; types - Ive mostly been following the original authors approach so didnt stop adding them until this PR - let me have a read through this link as maybe everyones wrong

Comment (BekYahia) [reply types/oracledb/index.d.ts]: Exactly 💯

**Comment by BekYahia** [comment]
LGTM!

**Review by afadul404 — APPROVED** [review]

**Review by sandersn — APPROVED** [review]

**Comment by dannyb648** [comment]
Ready to Merge

**Pull request merged**